### PR TITLE
Fix Travis-CI based deployment issues

### LINKED ISSUE #40: AWS deployment errors
There are a couple deployment errors after configuring the AWS related environment variables:

* `S3 client configured for "us-east-1" but the bucket "[secure]" is in "us-west-2"; Please configure the proper region to avoid multiple unnecessary redirects and signing attempts` when uploading to s3
* `OSError: [Errno 13] Permission denied: '/usr/local/lib/python2.7/dist-packages/dateutil'` when running "pip install awscli".

First one is more like a warning, but still worth fixing.

Please see https://travis-ci.org/mozilla-lockbox/mozilla-lockbox.github.io/jobs/358618355 for details.

### PR DESCRIPTION
**Pull request by linuxwolf**
Resolves #40 

This fixes the issue, but uses the completely different deployment setup.  There are too many issues with post-build installations in Travis-CI; instead this uses a deployment script with internal conditions.

### CODE REVIEW

File: _utils/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+fi
+
+SUPPORTED_BRANCH=${SUPPORTED_BRANCH:-master}
+if [[ ( "$TRAVIS" == "true" ) && \
+      ( "$TRAVIS_BRANCH" != "$SUPPORTED_BRANCH" )]]; then
+  echo "Travis-CI branch not supported; skipping deployment"
+  exit 0
+fi
+
+s3_website push
```
[the remark this answers is not in the document]
Comment (linuxwolf) [reply _utils/deploy.sh]: Unfortunately, no.  However, the console output it does generate has helped me diagnose problems more easily than I found with the `s3` provider.

[the remark this answers is not in the document]
Comment (bqbn) [reply _utils/deploy.sh]: Can you see from the console output that it indeed uploads to S3 us-west-2 region? We set the AWS_REGION to 'us-west-2', but from the travis-ci logs, I couldn't tell if that's the region it used.

[the remark this answers is not in the document]
Comment (linuxwolf) [reply _utils/deploy.sh]: when I _didn't_ have the region set, I saw warnings.  It's not ideal, but it did help diagnose issues with pushing the site.

[the remark this answers is not in the document]
Comment (bqbn) [reply _utils/deploy.sh]: OK, that's good news, which means now that there is **NOT** a warning about the region in travis-ci logs, the `AWS_REGION` environment variable is in effect and working.

**Pull request merged**